src/diffusers/models/unet_2d_condition.py

```diff
@@ -899,23 +899,35 @@ def forward(
         sample = self.conv_in(sample)
 
         # 3. down
+
+        is_controlnet = mid_block_additional_residual is not None and down_block_additional_residuals is not None
+        is_adapter = mid_block_additional_residual is None and down_block_additional_residuals is not None
+
         down_block_res_samples = (sample,)
         for downsample_block in self.down_blocks:
             if hasattr(downsample_block, "has_cross_attention") and downsample_block.has_cross_attention:
+                additional_kwargs = {}
```

Comment:
```suggestion
                additional_residuals = {}
```

Reply:
Ok to create dict and unpack directly here so that we don't have to adapt every downsample block. Could we add a comment here though saying that this is only needed for the CrossAttentionDownSampling block and only for t2i adapter?

Reply:
And would prefer to make it residual specific to not open the door to add whatever through this dict, want to make sure the reader understands that it's a special case

Reply:
added!